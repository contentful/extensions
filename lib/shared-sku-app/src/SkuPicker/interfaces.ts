export interface Pagination {
  count: number;
  limit: number;
  offset: number;
  total: number;
}
