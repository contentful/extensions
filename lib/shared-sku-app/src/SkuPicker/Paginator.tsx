import React from 'react';
import { Button } from '@contentful/forma-36-react-components';
import { css } from 'emotion';
import range from 'lodash/range';

interface Props {
  activePage: number;
  className?: string;
  pageCount: number;
  setActivePage: Function;
}

const styles = {
  button: css({
    borderRadius: 0,
    maxWidth: '45px',
    span: {
      overflow: 'visible !important'
    }
  })
};

function getPagesRange(page: number, total: number, neighboursCount = 2): number[] {
  if (total < neighboursCount * 2 + 1) {
    return range(0, total);
  }
  if (page <= neighboursCount) {
    return range(0, neighboursCount * 2 + 1);
  }
  if (page > total - neighboursCount) {
    return range(total - neighboursCount * 2 - 1, total);
  }
  return range(page - neighboursCount - 1, page + neighboursCount);
}

export function Paginator({ activePage, className, pageCount, setActivePage }: Props) {
  return (
    <div className={className}>
      <Button
        className={styles.button}
        buttonType="muted"
        disabled={pageCount === 1 || activePage === 1}
        onClick={() => setActivePage(1)}>
        &laquo;
      </Button>
      <Button
        icon="ChevronLeft"
        className={styles.button}
        buttonType="muted"
        disabled={pageCount === 1 || activePage === 1}
        onClick={() => setActivePage(activePage - 1)}
      />
      {getPagesRange(activePage, pageCount).map(pageIndex => (
        <Button
          onClick={() => setActivePage(pageIndex + 1)}
          className={styles.button}
          buttonType={pageIndex + 1 === activePage ? 'positive' : 'muted'}
          key={pageIndex}>
          {pageIndex + 1}
        </Button>
      ))}
      <Button
        icon="ChevronRight"
        buttonType="muted"
        className={styles.button}
        disabled={pageCount === 1 || activePage === pageCount}
        onClick={() => setActivePage(activePage + 1)}
      />
      <Button
        buttonType="muted"
        className={styles.button}
        disabled={pageCount === 1 || activePage === pageCount}
        onClick={() => setActivePage(pageCount)}>
        &raquo;
      </Button>
    </div>
  );
}
