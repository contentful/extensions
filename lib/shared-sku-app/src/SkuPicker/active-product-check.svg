<svg width="44" height="44" xmlns="http://www.w3.org/2000/svg"><g fill="#3072BE" fill-rule="evenodd"><circle fill-opacity=".1" cx="22" cy="22" r="22"/><path d="M31.689 16.614L29.555 14.5 19.024 25.275l-4.62-4.406-2.092 2.153 6.75 6.478 12.627-12.886z" fill-rule="nonzero"/></g></svg>