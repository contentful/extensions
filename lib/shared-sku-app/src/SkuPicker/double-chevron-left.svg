<svg width="18" height="18" xmlns="http://www.w3.org/2000/svg"><g fill="none" fill-rule="evenodd"><path d="M0 0h18v18H0z"/><path d="M13.221 4.5L8.721 9l4.5 4.5 1.058-1.057L10.844 9l3.435-3.442L13.22 4.5zm-5 0L3.721 9l4.5 4.5 1.058-1.057L5.844 9l3.435-3.442L8.22 4.5z" fill="#000" fill-rule="nonzero"/></g></svg>