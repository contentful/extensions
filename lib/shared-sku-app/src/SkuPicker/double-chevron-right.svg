<svg width="18" height="18" xmlns="http://www.w3.org/2000/svg"><g fill="none" fill-rule="evenodd"><path d="M0 0h18v18H0z"/><path d="M4.779 4.5l4.5 4.5-4.5 4.5-1.058-1.057L7.156 9 3.721 5.558 4.78 4.5zm5 0l4.5 4.5-4.5 4.5-1.058-1.057L12.156 9 8.721 5.558 9.78 4.5z" fill="#000" fill-rule="nonzero"/></g></svg>