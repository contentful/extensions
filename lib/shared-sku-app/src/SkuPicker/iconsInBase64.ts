export const activeProductCheck =
  'data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg==';

export const doubleChevron =
  'data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgiIGhlaWdodD0iMTgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj48cGF0aCBkPSJNMCAwaDE4djE4SDB6Ii8+PHBhdGggZD0iTTEzLjIyMSA0LjVMOC43MjEgOWw0LjUgNC41IDEuMDU4LTEuMDU3TDEwLjg0NCA5bDMuNDM1LTMuNDQyTDEzLjIyIDQuNXptLTUgMEwzLjcyMSA5bDQuNSA0LjUgMS4wNTgtMS4wNTdMNS44NDQgOWwzLjQzNS0zLjQ0Mkw4LjIyIDQuNXoiIGZpbGw9IiMwMDAiIGZpbGwtcnVsZT0ibm9uemVybyIvPjwvZz48L3N2Zz4=';
