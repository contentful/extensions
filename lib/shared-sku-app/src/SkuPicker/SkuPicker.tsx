import React, { Component } from 'react';
import get from 'lodash/get';
import clamp from 'lodash/clamp';
import debounce from 'lodash/debounce';
import { Button, TextInput, Icon } from '@contentful/forma-36-react-components';
import { DialogExtensionSDK } from 'contentful-ui-extensions-sdk';
import { Divider } from '../Divider';
import { ProductList } from './ProductList';
import { Paginator } from './Paginator';
import { Pagination, Product, Hash, ProductPreviewsFn, ProductsFn } from '../interfaces';
import { ProductSelectionList } from './ProductSelectionList';
import { styles } from './styles';
import { mapSort } from '../utils';

export interface Props {
  sdk: DialogExtensionSDK;
  fetchProductPreviews: ProductPreviewsFn;
  fetchProducts: ProductsFn;
}

interface State {
  activePage: number;
  search: string;
  pagination: Pagination;
  products: Product[];
  selectedProducts: Product[];
  selectedSKUs: string[];
}

const SEARCH_DELAY = 250;

function getSaveBtnText(selectedSKUs: string[]): string {
  switch (selectedSKUs.length) {
    case 0:
      return 'Save products';
    case 1:
      return 'Save 1 product';
    default:
      return `Save ${selectedSKUs.length} products`;
  }
}

export class SkuPicker extends Component<Props, State> {
  state: State = {
    activePage: 1,
    search: '',
    pagination: {
      count: 0,
      limit: 0,
      offset: 0,
      total: 0
    },
    products: [],
    selectedProducts: [],
    selectedSKUs: get(this.props, ['sdk', 'parameters', 'invocation', 'fieldValue'], [])
  };

  componentDidMount() {
    this.updateProducts();
    this.updateSelectedProducts();
  }

  setSearchCallback = debounce(() => {
    this.setActivePage(1);
    this.updateProducts();
  }, SEARCH_DELAY);

  setSearch = (search: string) => {
    this.setState({ search }, this.setSearchCallback);
  };

  updateProducts = async () => {
    try {
      const {
        activePage,
        pagination: { limit },
        search
      } = this.state;
      const offset = (activePage - 1) * limit;
      const { pagination, products } = await this.props.fetchProducts(search, { offset });
      this.setState({ pagination, products });
    } catch (error) {
      this.props.sdk.notifier.error('There was an error fetching the product list.');
    }
  };

  updateSelectedProducts = async () => {
    try {
      const { selectedSKUs } = this.state;
      const config = this.props.sdk.parameters.installation;
      const selectedProductsUnsorted = await this.props.fetchProductPreviews(selectedSKUs, config);
      const selectedProducts = mapSort(selectedProductsUnsorted, selectedSKUs, 'sku');
      this.setState({ selectedProducts });
    } catch (error) {
      this.props.sdk.notifier.error(
        'There was an error fetching the data for the selected products.'
      );
    }
  };

  setActivePage = (activePage: number) => {
    const { pagination } = this.state;
    const pageCount = Math.ceil(pagination.total / pagination.limit);
    this.setState({ activePage: clamp(activePage, 1, pageCount) }, () => {
      this.updateProducts();
    });
  };

  selectProduct = (sku: string) => {
    const { fieldType } = this.props.sdk.parameters.invocation as Hash;
    const onlyOneProductCanBeSelected = fieldType === 'Symbol';

    if (this.state.selectedSKUs.includes(sku)) {
      this.setState(
        ({ selectedSKUs }) => ({
          selectedSKUs: selectedSKUs.filter(productSku => productSku !== sku)
        }),
        () => this.updateSelectedProducts()
      );
    } else {
      this.setState(
        ({ selectedSKUs }) => ({
          selectedSKUs: onlyOneProductCanBeSelected ? [sku] : [...selectedSKUs, sku]
        }),
        () => this.updateSelectedProducts()
      );
    }
  };

  render() {
    const { search, pagination, products, selectedProducts, selectedSKUs } = this.state;
    const pageCount = Math.ceil(pagination.total / pagination.limit);

    return (
      <>
        <header className={styles.header}>
          <div className={styles.leftsideControls}>
            <TextInput
              placeholder="Search for a product..."
              type="search"
              name="sku-search"
              id="sku-search"
              testId="sku-search"
              width="medium"
              value={search}
              onChange={event => this.setSearch((event.target as HTMLInputElement).value)}
            />
            <Icon color="muted" icon="Search" />
            <span className={styles.total}>Total results: {pagination.total.toLocaleString()}</span>
          </div>
          <div className={styles.rightsideControls}>
            <ProductSelectionList products={selectedProducts} selectProduct={this.selectProduct} />
            <Button
              className={styles.saveBtn}
              buttonType="primary"
              onClick={() => this.props.sdk.close(selectedSKUs)}
              disabled={selectedSKUs.length === 0}>
              {getSaveBtnText(selectedSKUs)}
            </Button>
          </div>
        </header>
        <Divider />
        <section className={styles.body}>
          <ProductList
            products={products}
            selectProduct={this.selectProduct}
            selectedSKUs={selectedSKUs}
          />
          {products.length > 0 && (
            <Paginator
              activePage={this.state.activePage}
              className={styles.paginator}
              pageCount={pageCount}
              setActivePage={this.setActivePage}
            />
          )}
        </section>
      </>
    );
  }
}
