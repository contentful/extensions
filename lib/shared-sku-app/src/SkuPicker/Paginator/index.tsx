import React from 'react';
import { Button } from '@contentful/forma-36-react-components';
import clamp from 'lodash/clamp';
import range from 'lodash/range';
import { doubleChevron } from '../iconsInBase64';
import { styles } from './styles';

export interface Props {
  activePage: number;
  className?: string;
  pageCount: number;
  setActivePage: (page: number) => void;
}

export function getPagesRange(page: number, total: number, neighboursCount = 2): number[] {
  const PAGINATOR_RANGE = neighboursCount * 2;

  if (total <= PAGINATOR_RANGE) {
    // Total amount of pages are less than the possible paginator range
    return range(0, total);
  }
  if (page <= neighboursCount) {
    // Active page is at the start of the paginator page count
    return range(0, PAGINATOR_RANGE + 1);
  }
  if (page > total - neighboursCount) {
    // Active page is at the end of the paginator page count
    return range(total - PAGINATOR_RANGE - 1, total);
  }
  // Active page is in the middle of the paginator count
  return range(page - neighboursCount - 1, page + neighboursCount);
}

export function Paginator(props: Props) {
  const { className, pageCount, setActivePage } = props;
  const activePage = clamp(props.activePage, 1, pageCount);
  const hasOnlyOnePage = pageCount === 1;
  const activePageIsAtPaginatorStart = activePage === 1;
  const activePageIsAtPaginatorEnd = activePage === pageCount;

  return (
    <div className={className}>
      <Button
        className={styles.button}
        buttonType="muted"
        disabled={hasOnlyOnePage || activePageIsAtPaginatorStart}
        onClick={() => setActivePage(1)}>
        <img className={styles.chevronLeft} src={doubleChevron} alt="right" />
      </Button>
      <Button
        icon="ChevronLeft"
        className={styles.button}
        buttonType="muted"
        disabled={hasOnlyOnePage || activePageIsAtPaginatorStart}
        onClick={() => setActivePage(activePage - 1)}
      />
      {getPagesRange(activePage, pageCount).map(pageIndex => {
        const page = pageIndex + 1;
        return (
          <Button
            onClick={() => setActivePage(page)}
            className={styles.button}
            buttonType={page === activePage ? 'primary' : 'muted'}
            testId={page === activePage ? 'active' : `inactive-${page}`}
            key={pageIndex}>
            {page}
          </Button>
        );
      })}
      <Button
        icon="ChevronRight"
        buttonType="muted"
        className={styles.button}
        disabled={hasOnlyOnePage || activePageIsAtPaginatorEnd}
        onClick={() => setActivePage(activePage + 1)}
      />
      <Button
        buttonType="muted"
        className={styles.button}
        disabled={hasOnlyOnePage || activePageIsAtPaginatorEnd}
        onClick={() => setActivePage(pageCount)}>
        <img className={styles.chevronRight} src={doubleChevron} alt="right" />
      </Button>
    </div>
  );
}
