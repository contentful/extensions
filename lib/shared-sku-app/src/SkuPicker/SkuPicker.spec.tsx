import React from 'react';
import identity from 'lodash/identity';
import merge from 'lodash/merge';
import { fireEvent, configure, render, cleanup } from '@testing-library/react';
import { Props, SkuPicker } from './SkuPicker';
import productPreviews from '../__mocks__/productPreviews';
import { DialogExtensionSDK } from 'contentful-ui-extensions-sdk';

configure({
  testIdAttribute: 'data-test-id'
});

const defaultProps: Props = {
  sdk: ({
    parameters: {
      installation: {},
      invocation: {
        fieldType: 'Symbol',
        fieldValue: []
      }
    },
    close: jest.fn(),
    notifier: {
      success: jest.fn(),
      error: jest.fn()
    }
  } as unknown) as DialogExtensionSDK,
  fetchProductPreviews: jest.fn(skus =>
    productPreviews.filter(preview => skus.includes(preview.sku))
  ) as any,
  fetchProducts: jest.fn(() => ({
    pagination: {
      count: 3,
      limit: 20,
      offset: 0,
      total: 3
    },
    products: productPreviews
  })) as any
};

const renderComponent = async (props: Props) => {
  const component = render(<SkuPicker {...props} />);
  // wait for data to load and render
  await component.findByText('Dress Twin-Set rose');
  component.getAllByTestId('image').forEach(img => fireEvent(img, new Event('load')));
  return component;
};

jest.mock('react-sortable-hoc', () => ({
  SortableContainer: identity,
  SortableElement: identity,
  SortableHandle: identity
}));

describe('SkuPicker', () => {
  afterEach(cleanup);

  it('should render successfully with no products selected', async () => {
    const { container } = await renderComponent(defaultProps);
    expect(container).toMatchSnapshot();
  });

  describe('when it is operating on a field of type Symbol', () => {
    it('should allow the user to select only one product', async () => {
      const { queryByTestId, findByTestId } = await renderComponent(defaultProps);
      const productA = await findByTestId(`product-preview-${productPreviews[1].sku}`);
      const productB = await findByTestId(`product-preview-${productPreviews[2].sku}`);
      productA.click();
      productB.click();
      expect(queryByTestId(`selection-preview-${productPreviews[1].sku}`)).toBeNull();
      expect(await findByTestId(`selection-preview-${productPreviews[2].sku}`)).toBeTruthy();
    });
  });

  describe('when it is operating on a field of type Symbol[]', () => {
    it('should allow the user to select multiple products', async () => {
      const { findByTestId } = await renderComponent({
        ...defaultProps,
        sdk: merge({}, defaultProps.sdk, {
          parameters: { invocation: { fieldType: 'Array' } }
        })
      });
      const productA = await findByTestId(`product-preview-${productPreviews[1].sku}`);
      const productB = await findByTestId(`product-preview-${productPreviews[2].sku}`);
      productA.click();
      productB.click();
      expect(await findByTestId(`selection-preview-${productPreviews[1].sku}`)).toBeTruthy();
      expect(await findByTestId(`selection-preview-${productPreviews[2].sku}`)).toBeTruthy();
    });
  });
});
