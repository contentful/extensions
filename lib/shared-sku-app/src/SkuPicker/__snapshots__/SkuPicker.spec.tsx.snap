// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkuPicker should render successfully with no products selected 1`] = `
<div>
  <header
    class="css-ss1rwp"
  >
    <div
      class="css-kpva4h"
    >
      <div
        class="TextInput__TextInput___36-K- TextInput__TextInput--medium___1bR2D"
      >
        <input
          aria-label="sku-search"
          class="TextInput__TextInput__input___27vDB a11y__focus-border--default___60AXp"
          data-test-id="sku-search"
          id="sku-search"
          name="sku-search"
          placeholder="Search for a product..."
          type="search"
          value=""
        />
      </div>
      <svg
        class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
        data-test-id="cf-ui-icon"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0 0 16 9.5 6.5 6.5 0 1 0 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <span
        class="css-1nagyg6"
      >
        Total results: 
        3
      </span>
    </div>
    <div
      class="css-kb3y9n"
    >
      <div
        class="css-k008qs"
      />
      <button
        class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-1kr4fmq Button__Button--primary___JImeO Button__Button--disabled___1E20M"
        data-test-id="cf-ui-button"
        disabled=""
        type="button"
      >
        <span
          class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
          tabindex="-1"
        >
          <span
            class="Button__Button__label___3tcOj"
          >
            Save products
          </span>
        </span>
      </button>
    </div>
  </header>
  <hr
    class="css-6wso32"
  />
  <section
    class="css-gj69t3"
  >
    <div
      class="css-almiqb"
    >
      <div
        class="css-zkv16y"
      >
        <div
          aria-checked="false"
          class="css-1qso258 "
          data-test-id="product-preview-M0E20130820E90Z"
          role="switch"
          tabindex="-1"
        >
          <div
            class="css-19gid52"
          >
            <img
              alt="product preview"
              class="css-1ddetr"
              data-test-id="image"
              src="https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-1.jpg"
              style="display: block;"
            />
            <img
              alt="check"
              class="css-1imdisq"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
            />
          </div>
          <p
            class="css-182natc"
          >
            Dress Twin-Set rose
          </p>
          <p
            class="css-1kn9x3e"
          >
            M0E20130820E90Z
          </p>
        </div>
      </div>
      <div
        class="css-zkv16y"
      >
        <div
          aria-checked="false"
          class="css-1qso258 "
          data-test-id="product-preview-A0E2300FX102203"
          role="switch"
          tabindex="-1"
        >
          <div
            class="css-19gid52"
          >
            <img
              alt="product preview"
              class="css-1ddetr"
              data-test-id="image"
              src="https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-2.jpg"
              style="display: block;"
            />
            <img
              alt="check"
              class="css-1imdisq"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
            />
          </div>
          <p
            class="css-182natc"
          >
            Michael Kors – Shopper “Jet Set Travel”
          </p>
          <p
            class="css-1kn9x3e"
          >
            A0E2300FX102203
          </p>
        </div>
      </div>
      <div
        class="css-zkv16y"
      >
        <div
          aria-checked="false"
          class="css-1qso258 "
          data-test-id="product-preview-M0E21300900DZN7"
          role="switch"
          tabindex="-1"
        >
          <div
            class="css-19gid52"
          >
            <img
              alt="product preview"
              class="css-1ddetr"
              data-test-id="image"
              src="https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-3.jpg"
              style="display: block;"
            />
            <img
              alt="check"
              class="css-1imdisq"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
            />
          </div>
          <p
            class="css-182natc"
          >
            Shirt “Jenny“ Polo Ralph Lauren white
          </p>
          <p
            class="css-1kn9x3e"
          >
            M0E21300900DZN7
          </p>
        </div>
      </div>
    </div>
    <div
      class="css-1twx1i5"
    >
      <button
        class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-fedt8m Button__Button--muted___2Wair Button__Button--disabled___1E20M"
        data-test-id="cf-ui-button"
        disabled=""
        type="button"
      >
        <span
          class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
          tabindex="-1"
        >
          <span
            class="Button__Button__label___3tcOj"
          >
            <img
              alt="right"
              class="css-1t0k538"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgiIGhlaWdodD0iMTgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj48cGF0aCBkPSJNMCAwaDE4djE4SDB6Ii8+PHBhdGggZD0iTTEzLjIyMSA0LjVMOC43MjEgOWw0LjUgNC41IDEuMDU4LTEuMDU3TDEwLjg0NCA5bDMuNDM1LTMuNDQyTDEzLjIyIDQuNXptLTUgMEwzLjcyMSA5bDQuNSA0LjUgMS4wNTgtMS4wNTdMNS44NDQgOWwzLjQzNS0zLjQ0Mkw4LjIyIDQuNXoiIGZpbGw9IiMwMDAiIGZpbGwtcnVsZT0ibm9uemVybyIvPjwvZz48L3N2Zz4="
            />
          </span>
        </span>
      </button>
      <button
        class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-fedt8m Button__Button--muted___2Wair Button__Button--disabled___1E20M"
        data-test-id="cf-ui-button"
        disabled=""
        type="button"
      >
        <span
          class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
          tabindex="-1"
        >
          <svg
            class="Icon__Icon___38Epv Button__Button__icon___2YX5- Icon__Icon--small___1yGZK Icon__Icon--secondary___1ztcw"
            data-test-id="cf-ui-icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
      </button>
      <button
        class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-fedt8m Button__Button--primary___JImeO"
        data-test-id="active"
        type="button"
      >
        <span
          class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
          tabindex="-1"
        >
          <span
            class="Button__Button__label___3tcOj"
          >
            1
          </span>
        </span>
      </button>
      <button
        class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-fedt8m Button__Button--muted___2Wair Button__Button--disabled___1E20M"
        data-test-id="cf-ui-button"
        disabled=""
        type="button"
      >
        <span
          class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
          tabindex="-1"
        >
          <svg
            class="Icon__Icon___38Epv Button__Button__icon___2YX5- Icon__Icon--small___1yGZK Icon__Icon--secondary___1ztcw"
            data-test-id="cf-ui-icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
      </button>
      <button
        class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-fedt8m Button__Button--muted___2Wair Button__Button--disabled___1E20M"
        data-test-id="cf-ui-button"
        disabled=""
        type="button"
      >
        <span
          class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
          tabindex="-1"
        >
          <span
            class="Button__Button__label___3tcOj"
          >
            <img
              alt="right"
              class="css-11hf54p"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgiIGhlaWdodD0iMTgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj48cGF0aCBkPSJNMCAwaDE4djE4SDB6Ii8+PHBhdGggZD0iTTEzLjIyMSA0LjVMOC43MjEgOWw0LjUgNC41IDEuMDU4LTEuMDU3TDEwLjg0NCA5bDMuNDM1LTMuNDQyTDEzLjIyIDQuNXptLTUgMEwzLjcyMSA5bDQuNSA0LjUgMS4wNTgtMS4wNTdMNS44NDQgOWwzLjQzNS0zLjQ0Mkw4LjIyIDQuNXoiIGZpbGw9IiMwMDAiIGZpbGwtcnVsZT0ibm9uemVybyIvPjwvZz48L3N2Zz4="
            />
          </span>
        </span>
      </button>
    </div>
  </section>
</div>
`;
